### PR TITLE
Feature/get dog by

### PR DESCRIPTION
**Pull request by Aska0526**

### CODE REVIEW

File: java-api/src/test/java/com/db/grad/javaapi/service/DogHandlerTest.java [hunk 1]
```diff
@@ -55,4 +56,53 @@ public void add_several_dog_return_number_of_dogs_match_number_added(){
...
+    public void find_dog_by_wrong_id_returns_null(){
+
+        DogHandler cut = new DogHandler(itsDogRepo);
+
+        Dog theDog = new Dog();
+        theDog.setName("Bruce");
+        cut.addDog(theDog);
+        theDog = new Dog();
+        theDog.setName("Clark");
+        long uniqueID = cut.addDog(theDog);
```
Comment (salkaevruslan) [inline java-api/src/test/java/com/db/grad/javaapi/service/DogHandlerTest.java]: uniqueID is never used

File: java-api/src/test/java/com/db/grad/javaapi/service/DogHandlerTest.java [hunk 2]
```diff
@@ -55,4 +56,53 @@ public void add_several_dog_return_number_of_dogs_match_number_added(){
...
+        Dog expectedDog = theDog;
```
Comment (salkaevruslan) [inline java-api/src/test/java/com/db/grad/javaapi/service/DogHandlerTest.java]: Unused

**Review by salkaevruslan — CHANGES_REQUESTED** [review]

File: java-api/src/test/java/com/db/grad/javaapi/service/DogHandlerTest.java [hunk 3]
```diff
@@ -55,4 +56,42 @@ public void add_several_dog_return_number_of_dogs_match_number_added(){
...
+
+        assertEquals(expectedDog.getId(),actualResult.getId());
+        assertEquals(expectedDog.getName(),actualResult.getName());
+
+    }
+
+    @Test
+    public void find_dog_by_wrong_id_returns_null(){
+
+        DogHandler cut = new DogHandler(itsDogRepo);
```
Comment (salkaevruslan) [inline java-api/src/test/java/com/db/grad/javaapi/service/DogHandlerTest.java:139]: Maybe add 1 dog here to show that the reason for null is not an empty repo.

**Review by salkaevruslan — APPROVED** [review]
LGTM, you can see 1 more comment, but not is also ok.

**Review by vlados2000 — COMMENTED** [review]
nice

**Pull request merged**